gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -133,30 +133,44 @@ class {{ service.name }}RestTransport({{ service.name }}Transport):
         {%- endif %}
         """
 
-        {%- if 'body' in method.http_opt.keys() %}
+        {%- if 'body' in method.http_opt %}
         # Jsonify the input
-        data = {{ method.output.ident }}.to_json(
-            {%- if method.http_opt['body'] == '*' %}
+        {%- if method.http_opt['body'] != '*' %}
+        data = {{ method.input.fields[method.http_opt['body']].type.ident }}.to_json(
+            request.{{ method.http_opt['body'] }},
+            including_default_value_fields=False
+        )
+        {%- else %}
+        data = {{ method.input.ident }}.to_json(
             request
-            {%- else %}
-            request.body
-            {%- endif %}
         )
         {%- endif %}
+        {%- endif %}
 
         {# TODO(yon-mg): Write helper method for handling grpc transcoding url #}
         # TODO(yon-mg): need to handle grpc transcoding and parse url correctly
-        #              current impl assumes simpler version of grpc transcoding
+        #               current impl assumes basic case of grpc transcoding
         # Send the request
         url = 'https://{host}{{ method.http_opt['url'] }}'.format(
             host=self._host,
-            {%- for field in method.input.fields.keys() %}
+            {%- for field in method.path_params %}
             {{ field }}=request.{{ field }},
             {%- endfor %}
         )
+
+        potentialParams = {
+            {%- for field in method.query_params %}
+            '{{ field|camel_case }}': request.{{ field }},
+            {%- endfor %}
+        }
+        potentialParams = ((k, v) for k, v in potentialParams.items() if v)
+        for i, (param_name, param_value) in enumerate(potentialParams):
+            q = '?' if i == 0 else '&'
+            url += "{q}{name}={value}".format(q=q, name=param_name, value=param_value.replace(' ', '+'))
```

Comment:
!! Does this ensure values are URL-encoded? If not, at least add a TODO.

Reply:
Good point! Will do that

Reply:
So the short answer is yes but I will add tests to confirm that.

Reply:
OK, great. If it's not trivial to URL encode, you can leave a TODO and do it in a follow-up PR, which is also fine. That would let us start to use/test what you have here for well formed data where URL encoding isn't necessary.

Reply:
(BTW, do you know why the `&` shows up in red on GitHub?)

Reply:
Not sure. Seems to be a bug w/ ampersand messing with syntax highlighting? Certainly no real error AFAIK.

Reply:
Thanks for the TODO!